assignment_5.1_web_scraping.py

```diff
@@ -0,0 +1,229 @@
+'''
+Importing the necessary packages for web scraping
+'''
+from bs4 import BeautifulSoup
+import requests 
+import pandas as pd
+
+class WebScraping:
+
+    def __init__(self):
+        pass
+    
+    def get_html_content(self):        
+        '''
+        This function uses the requests module to get the html contents of the web page
+        '''
+        url='https://www.flipkart.com/search?q=laptops&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=on&as=off'
+        try:
+            self.response=requests.get(url)
+            self.response.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            return 'Error occured',e
+        
+    def get_product_name(self):
+        '''
+        This function is used to get the product name of all the laptops in the web page
+        '''
+        self.soup=BeautifulSoup(self.response .content,'html.parser')
```

Comment:
self.response (you should not have space here) it should be self.response.content

Reply:
ok sir